scripts/utils/termux_reuse_pr_build_artifacts.sh

```diff
@@ -0,0 +1,125 @@
+
+# We should obtain data about all commits in this push and make sure they all came from the same PR if any.
+# No hard tabs or newlines allowed inside a graphql request.
+GRAPHQL_QUERY=$(tr '\t\n' '  ' <<< "
+query {
+  repository(owner: \\\"termux\\\", name: \\\"termux-packages\\\") {
+  $(
+    for i in $(git rev-list --no-merges "$OLD_COMMIT..$HEAD_COMMIT" || true); do
+      echo "_${i:0:7}: object(oid: \\\"${i}\\\") { ...commitWithPR }"
+    done
+  )
+  }
+}
+
+fragment commitWithPR on Commit {
+  associatedPullRequests(first: 1) { nodes { baseRefOid headRefOid } edges { node { number } } }
+}")
+
+# Send graphql to github and obtain response
+RESPONSE="$(
+curl -s \
+  -H "Authorization: bearer $GITHUB_TOKEN" \
+  -H "Content-Type: application/json" \
+  -X POST -d "{ \"query\": \"$GRAPHQL_QUERY\"}" \
+  https://api.github.com/graphql
+)" || error_exit "Failed to perform GraphQL request, not performing CI fast path"
+
+# Ensure response is valid and obtain all associated PR numbers if any and make sure response is valid
+PRS="$(jq '.data.repository | to_entries[] | .value.associatedPullRequests.edges
+         | (.[] // []) | (.node? // []) | (.number? // 0)' <<< "$RESPONSE")" || \
+         error_exit "GraphQL response is invalid, not performing CI fast path"
+
+# Check if all commits come from the same PR and return if not
+[ "$(sort -u <<< "$PRS" | wc -l)" -eq 0 ] && error_exit "push does not contain commits from PR, not performing CI fast path"
+[ "$(sort -u <<< "$PRS" | wc -l)" -eq 1 ] || error_exit "push contains commits from more than one pull request, not performing CI fast path"
+
+PR_BASE_COMMIT="$(jq -r '.data.repository | to_entries[0].value.associatedPullRequests.nodes[0].baseRefOid' <<< "$RESPONSE")" || exit 0
+PR_HEAD_COMMIT="$(jq -r '.data.repository | to_entries[0].value.associatedPullRequests.nodes[0].headRefOid' <<< "$RESPONSE")" || exit 0
+[ -n "${PR_BASE_COMMIT:-}" ] || error_exit "failed to read associated PR base commit, not performing CI fast path"
+[ -n "${PR_HEAD_COMMIT:-}" ] || error_exit "failed to read associated PR head commit, not performing CI fast path"
+
+DIRS_REGEX="$(echo "$TERMUX_PACKAGE_DIRECTORIES" | paste -sd'|' -)" || exit 0
+
+# fetch PR commit tree
+git fetch origin $PR_BASE_COMMIT:ref/tmp/$PR_BASE_COMMIT || error_exit "failed to fetch PR base tree, not performing CI fast path"
+git fetch origin $PR_HEAD_COMMIT:ref/tmp/$PR_HEAD_COMMIT || error_exit "failed to fetch PR head tree, not performing CI fast path"
+
+# make sure nobody force-pushed after CI was invoked and before we fetched remote tree and requested GraphQL data
+# checking if changed files in PR are same as in this push should be enough.
+
+diff -q \
+    <(git diff $PR_BASE_COMMIT $PR_HEAD_COMMIT | sed -n -E '/^diff --git a\// { p; n; /^index /!p; b } ; p') \
+    <(git diff $OLD_COMMIT $HEAD_COMMIT | sed -n -E '/^diff --git a\// { p; n; /^index /!p; b } ; p') >/dev/null 2>&1 \
```

Comment:
This is gonna need some explanation.
Especially the `sed` calls.

Also quote yo vars.

```suggestion
    <(git diff "$PR_BASE_COMMIT" "$PR_HEAD_COMMIT" | sed -n -E '/^diff --git a\// { p; n; /^index /!p; b } ; p') \
    <(git diff "$OLD_COMMIT" "$HEAD_COMMIT" | sed -n -E '/^diff --git a\// { p; n; /^index /!p; b } ; p') >/dev/null 2>&1 \
```